### DISCUSSION TITLE
Provide analytics for Org CTA

### DISCUSSION THREAD

**Discussion by michael-tharrington** [forem/forem#17288] [Feature Requests]
## Background

We have a variety of organization analytics that can be viewed from a user's dashboard (if they're part of an org). They can view stats specific to an article (see fig 1) or look at an overview of analytics for the whole org (see fig 2).

Orgs can also set a "Call to Action Box" (see fig 3) that appears on any post shared under the org (see fig 4). 

## Request

It'd be helpful if we could capture the stats on how many clicks the URL included in the call to action box has received and present that info in both per-article stats and overview stats for the org on the whole.

### Fig 1
![Screen Shot 2022-04-14 at 9 57 23 AM](https://user-images.githubusercontent.com/16007075/163406283-46fd8a6a-1ebe-4469-8079-be01ea04380e.png)

### Fig 2
![Screen Shot 2022-04-14 at 9 56 44 AM](https://user-images.githubusercontent.com/16007075/163406243-8d510335-9950-449d-9a24-ddc12dc61517.png)

### Fig 3
![Screen Shot 2022-04-14 at 9 59 58 AM](https://user-images.githubusercontent.com/16007075/163406773-0b4d9038-2433-4900-8eef-298cbce20366.png)

### Fig 4
![Screen Shot 2022-04-14 at 10 00 12 AM](https://user-images.githubusercontent.com/16007075/163406885-60cdcd41-5a93-43aa-868f-964b50d6b434.png)

**Comment by ellativity** [comment]
I'm wondering how much utility this will have to orgs, compared to them setting their own UTM link under `/settings/organization` > **Organization Details** > **Call-to-action box** 

I feel like it's one thing providing internal analytics, but tracking engagement with external links is outside of the scope of Forem's analytics. Happy to discuss this further if you want!

**Reply by michael-tharrington** [reply]
Totally hear ya! It would be possible for an org admin to add a UTM link.

That said, I recently spoke to an org admin and suggested a UTM, but they noted that they would've liked to have been able to track if the click on the CTA was connected to a specific article. They really wanted to see it listed in the specific article's `/stats` page so that they could tell if that particular article resonated and caused the community member to click on their CTA.

But I also understand what you mean about tracking engagement with external links being outside of the scope of Forem's analytics. The current level of analytics provided is extensive and we've selected it in such a way that we feel it promotes healthy interaction between orgs and our community members, so we want to be careful about whatever else we may add into the mix.
